From 4f25d5d09e9f73312d4d20a652c9ff214a66ead3 Mon Sep 17 00:00:00 2001
From: Robin Gong <b38343@freescale.com>
Date: Tue, 25 Aug 2015 14:25:22 +0800
Subject: [PATCH 0625/5242] MLK-11415: ARM: dts: fix the pinctrl conflict on
 i.MX7D 12x12 lpddr3 ARM2 board

commit  45e0eaa8370b357c4289db4561fca5624dedb7ac from
https://source.codeaurora.org/external/imx/linux-imx.git

Signed-off-by: Robin Gong <b38343@freescale.com>
Signed-off-by: Han Xu <b45815@freescale.com>

Also squashed the following:
MLK-11409-1: ARM: dts: imx7d-12x12-lpddr3-arm2.dts
Restore pinhogs after exit from lpsr mode.

It otherwise didn't build

Signed-off-by: Leonard Crestez <leonard.crestez@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/boot/dts/imx7d-12x12-lpddr3-arm2-sai.dts |    4 +---
 arch/arm/boot/dts/imx7d-12x12-lpddr3-arm2.dts     |    5 +++--
 2 files changed, 4 insertions(+), 5 deletions(-)

diff --git a/arch/arm/boot/dts/imx7d-12x12-lpddr3-arm2-sai.dts b/arch/arm/boot/dts/imx7d-12x12-lpddr3-arm2-sai.dts
index f1d36ce..9b0d635 100644
--- a/arch/arm/boot/dts/imx7d-12x12-lpddr3-arm2-sai.dts
+++ b/arch/arm/boot/dts/imx7d-12x12-lpddr3-arm2-sai.dts
@@ -21,9 +21,7 @@
 };
 
 &iomuxc {
-	pinctrl-names = "default", "sleep";
-	pinctrl-0 = <&pinctrl_hog_1 &pinctrl_hog_headphone_det>;
-	pinctrl-1 = <&pinctrl_hog_1 &pinctrl_hog_sd2_vselect>;
+	pinctrl-0 = <&pinctrl_hog_1>;
 };
 
 &sai1 {
diff --git a/arch/arm/boot/dts/imx7d-12x12-lpddr3-arm2.dts b/arch/arm/boot/dts/imx7d-12x12-lpddr3-arm2.dts
index c07a113..d123524 100644
--- a/arch/arm/boot/dts/imx7d-12x12-lpddr3-arm2.dts
+++ b/arch/arm/boot/dts/imx7d-12x12-lpddr3-arm2.dts
@@ -423,8 +423,9 @@
 };
 
 &iomuxc {
-	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_hog_1 &pinctrl_hog_sd2_vselect>;
+	pinctrl-names = "default", "sleep";
+	pinctrl-0 = <&pinctrl_hog_1 &pinctrl_hog_sd2_vselect &pinctrl_hog_mipi>;
+	pinctrl-1 = <&pinctrl_hog_1 &pinctrl_hog_sd2_vselect &pinctrl_hog_mipi>;
 
 	imx7d-12x12-lpddr3-arm2 {
 
-- 
1.7.9.5

